### PR TITLE
post three type discount and get store discount by storeID

### LINKED ISSUE #64: 編輯折扣

### PR DESCRIPTION
**Pull request by wj461**

### CODE REVIEW

File: backend/discount/delivery/discount_handler.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+func NewDiscountHandler(e *gin.Engine, discountUsecase domain.DiscountUsecase) {
+	handler := &DiscountHandler{
+		DiscountUsecase: discountUsecase,
+	}
+	e.GET("/api/v1/store/:storeID/discounts", handler.GetAllDiscountByStoreId)
+	e.POST("/api/v1/store/:storeID/seasoning-discount", handler.AddSeasoningDiscount)
+	e.POST("/api/v1/store/:storeID/shipping-discount", handler.AddShippingDiscount)
+	e.POST("/api/v1/store/:storeID/event-discount", handler.AddEventDiscount)
```
Comment (nonorze-VOXX) [inline backend/discount/delivery/discount_handler.go]: router := gin.Default()

	// Simple group: v1
	v1 := router.Group("/v1")
	{
		v1.POST("/login", loginEndpoint)
		v1.POST("/submit", submitEndpoint)
		v1.POST("/read", readEndpoint)
	}

File: backend/discount/delivery/discount_handler.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+type DiscountHandler struct {
+	DiscountUsecase domain.DiscountUsecase
+}
+
+func NewDiscountHandler(e *gin.Engine, discountUsecase domain.DiscountUsecase) {
+	handler := &DiscountHandler{
+		DiscountUsecase: discountUsecase,
+	}
+	e.GET("/api/v1/store/:storeID/discounts", handler.GetAllShippingByStoreId)
+	e.POST("/api/v1/seasoning-discount", handler.AddSeasoningDiscount)
```
Comment (nonorze-VOXX) [inline backend/discount/delivery/discount_handler.go]: no work by swagger post

File: backend/discount/delivery/discount_handler.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+	e.POST("/api/v1/store/:storeID/shipping-discount", handler.AddShippingDiscount)
+	e.POST("/api/v1/product/:productID/event-discount", handler.AddEventDiscount)
```
Comment (nonorze-VOXX) [inline backend/discount/delivery/discount_handler.go]: no work by swagger post

**Review by nonorze-VOXX — CHANGES_REQUESTED** [review]

File: backend/discount/delivery/discount_handler.go (hunk 3 above)
Comment (wj461) [reply backend/discount/delivery/discount_handler.go]: I use swagger post no problem?

**Review by nonorze-VOXX — DISMISSED** [review]

File: db/i35-shipping_discount.sql [hunk 4]
```diff
@@ -0,0 +1,6 @@
+INSERT INTO shipping_discount (discount_id, max_price, store_id) VALUES
+  (1,1,1),
+  (2,2,1),
+  (3,3,1),
+  (1,4,2),
+  (2,5,2)
```
Comment (huhuhu0420) [inline db/i35-shipping_discount.sql]: i think discount id should be unique

File: backend/discount/delivery/discount_handler.go (hunk 1 above)
Comment (huhuhu0420) [reply backend/discount/delivery/discount_handler.go]: why no do this

File: db/i35-shipping_discount.sql (hunk 4 above)
Comment (wj461) [reply db/i35-shipping_discount.sql]: fix

File: backend/discount/delivery/discount_handler.go (hunk 1 above)
Comment (wj461) [reply backend/discount/delivery/discount_handler.go]: fix

File: db/i35-shipping_discount.sql (hunk 4 above)
Comment (huhuhu0420) [reply db/i35-shipping_discount.sql]: why u think 1, 2, 3, 1, 2 are unique

Comment (wj461) [reply db/i35-shipping_discount.sql]: oh fix

File: backend/discount/delivery/discount_handler.go [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+}
+
+func (s *DiscountHandler) AddSeasoningDiscount(c *gin.Context) {
+	var discount swagger.SeasoningDiscount
+
+	if err := c.BindJSON(&discount); err != nil {
+		logrus.Error(err)
+		return
+	}
```
Comment (huhuhu0420) [inline backend/discount/delivery/discount_handler.go]: why this miss c.Status(400)
u need copilot

**Review by huhuhu0420 — COMMENTED** [review]
why u have fie changed  in swagger folder but there's no change in swagger.yaml

File: backend/discount/delivery/discount_handler.go (hunk 5 above)
Comment (wj461) [reply backend/discount/delivery/discount_handler.go]: fix

Comment (wj461) [reply backend/discount/delivery/discount_handler.go]: fix fix

**Review by huhuhu0420 — APPROVED** [review]

**Pull request merged**